clients/java/signalr/src/test/java/HubConnectionTest.java

```diff
@@ -19,6 +19,46 @@ public void checkHubConnectionState() throws InterruptedException {
         assertFalse(hubConnection.connected);
     }
 
+    @Test
+    public void RegisteringMultipleHandlersAndBothGetTriggered() throws Exception {
+
+        AtomicReference<Double> value = new AtomicReference<>(0.0);
+        Transport mockTransport = new MockEchoTransport();
+        HubConnection hubConnection = new HubConnection("http://example.com", mockTransport);
+        Action action = () -> value.getAndSet(value.get() + 1);
```

Comment:
I think `value.getAndSet(value.get() + x)` is the incorrect pattern, it should be `value.getAndUpdate((v) -> v + 1)`

Reply:
It just might look weird having a one line lambda inside of another one line lambda

Reply:
Yeah, try some multi-line lambdas, might look better